PL2/PL2-gui/src/main/java/nl/tudelft/pl2016gr2/gui/view/tree/ViewNode.java

```diff
@@ -64,62 +82,64 @@ public IPhylogeneticTreeNode getDataNode() {
   /**
    * Draw a root node and all of its children which fit on the screen.
    *
-   * @param root the root node.
-   * @param controller the controller of the tree.
+   * @param root             the root node.
+   * @param graphPane        the pane in which to draw the node.
+   * @param selectionManager the selection manager.
    * @return the nl.tudelft.pl2016gr2.gui.view node of the root.
    */
-  protected static ViewNode drawRootNode(IPhylogeneticTreeNode root, TreeController controller) {
-    Pane graphPane = controller.getGraphPane();
-    double startX = TreeController.GRAPH_BORDER_OFFSET;
-    double endX = graphPane.getWidth() - TreeController.GRAPH_BORDER_OFFSET;
-    double startY = TreeController.GRAPH_BORDER_OFFSET;
-    double endY = graphPane.getHeight() - TreeController.GRAPH_BORDER_OFFSET;
+  protected static ViewNode drawRootNode(IPhylogeneticTreeNode root, Pane graphPane,
+      SelectionManager selectionManager) {
+    double startX = TreeManager.GRAPH_BORDER_OFFSET;
+    double endX = graphPane.getWidth() - TreeManager.GRAPH_BORDER_OFFSET;
+    double startY = TreeManager.GRAPH_BORDER_OFFSET;
+    double endY = graphPane.getHeight() - TreeManager.GRAPH_BORDER_OFFSET;
     Area gbox = new Area(startX, endX, startY, endY);
-    return drawNode(root, gbox, controller);
+    return drawNode(root, gbox, graphPane, selectionManager);
   }
 
   /**
    * Recursively draw the node and all of its children.
    *
-   * @param dataNode the data of the node to draw.
-   * @param parent the parent of the node to draw.
-   * @param graphArea the area in which the node should be drawn.
-   * @param controller the controller of the tree.
+   * @param dataNode         the data of the node to draw.
+   * @param parent           the parent of the node to draw.
+   * @param graphArea        the area in which the node should be drawn.
+   * @param graphPane        the pane in which to draw the node.
+   * @param selectionManager the selection manager.
    * @return the drawn nl.tudelft.pl2016gr2.gui.view node.
    */
   private static ViewNode drawNode(IPhylogeneticTreeNode dataNode, Area graphArea,
-          TreeController controller) {
+      Pane graphPane, SelectionManager selectionManager) {
     if (graphArea.getWidth() < NODE_DIAMETER || graphArea.getHeight() < NODE_DIAMETER
-            || dataNode == null) {
+        || dataNode == null) {
       return null; // box too small to draw node.
     }
-    ViewNode node = new ViewNode(dataNode, graphArea);
-    controller.getGraphPane().getChildren().add(node);
+    ViewNode node = new ViewNode(dataNode, graphArea, selectionManager);
+    graphPane.getChildren().add(node);
```

Comment:
All right, as I said, not a JavaFX veteran

On Tue, 10 May 2016, 15:00 Faris Elghlan, <notifications@github.com> wrote:

> In
> PL2/PL2-gui/src/main/java/nl/tudelft/pl2016gr2/gui/view/tree/ViewNode.java
> <https://github.com/ProgrammingLife2016/PL2-2016/pull/57#discussion_r62666285>
> :
>
> >        return null; // box too small to draw node.
> >      }
> > -    ViewNode node = new ViewNode(dataNode, graphArea);
> > -    controller.getGraphPane().getChildren().add(node);
> > +    ViewNode node = new ViewNode(dataNode, graphArea, selectionManager);
> > +    graphPane.getChildren().add(node);
>
> graphPane is an instance of the JavaFX Pane class. All of JavaFXs classes
> which you can add child objects to work like this: "node.getsome
> observable list.add([stuff to add])".
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly or view it on GitHub
> <https://github.com/ProgrammingLife2016/PL2-2016/pull/57/files/ab9338aeaf5d18b7cd1188868b945280694173fd#r62666285>
>